### PR TITLE
JW-954 build cleanup

### PR DESCRIPTION
**Pull request by mclapinski**

### CODE REVIEW

File: rules.py [hunk 1]
```diff
@@ -617,17 +609,13 @@ def SandeshCppBuilder(target, source, env):
...
-    print '\nexecuting:' + '(cd '+ opath + ' & xxd -i ' + hname + ' >> ' + os.path.basename(cname) + ' )'
-    os.system("(cd " + opath + " & xxd -i " + hname + " >> " + os.path.basename(cname) + " )")
-    print 'after xxd, before cat'
-    os.system("echo } " + " >> " + cname)
-    os.system("cat " + tname + " >> " + cname)
-    print 'after cat'
-    print 'leave sandeshcppbuilder'
+    with open(cname, "a") as cfile:
+        cfile.write("namespace {")
+    subprocess.call("xxd -i " + hname + " >> " + os.path.basename(cname), shell = True, cwd = opath)
```
Comment (katrybacka) [inline rules.py]: remove spaces before and after equal sign

File: rules.py [hunk 2]
```diff
@@ -617,17 +609,13 @@ def SandeshCppBuilder(target, source, env):
...
+    with open(cname, "a") as cfile:
+        cfile.write("}")
+    subprocess.call("cat " + tname + " >> " + cname, shell = True)
```
Comment (katrybacka) [inline rules.py]: same here

File: rules.py [hunk 3]
```diff
@@ -762,12 +748,12 @@ def ThriftGenPyFunc(env, path, target=''):
...
         target += '/'
     targets = map(lambda module: target + 'gen_py/' + mod_dir + module, modules)
-    env.Depends(targets, '#/build/bin/thrift.exe')
+    env.Depends(targets, '#build/bin/thrift' + env['PROGSUFFIX'])
     return env.ThriftPy(targets, path)
 
 def IFMapBuilderCmd(source, target, env, for_signature):
     output = Basename(source[0].abspath)
-    return '\"tools/generateds/generateDS.py\" -f -g ifmap-backend -o %s %s' % (output, source[0])
+    return '%s -f -g ifmap-backend -o %s %s' % (os.path.join('.', 'tools', 'generateds', 'generateDS.py'), output, source[0])
```
Comment (katrybacka) [inline rules.py]: better use `File(#tools/generateds/generateDS.py).abspath`

File: rules.py (hunk 1 above)
Comment (katrybacka) [inline rules.py]: remove spaces before and after equal sign

File: rules.py (hunk 2 above)
Comment (katrybacka) [inline rules.py]: same here

File: rules.py [hunk 4]
```diff
@@ -784,7 +770,7 @@ def CreateIFMapBuilder(env):
 
 def TypeBuilderCmd(source, target, env, for_signature):
     output = Basename(source[0].abspath)
-    return '"tools/generateds/generateDS.py" -f -g type -o %s %s' % (output, source[0])
+    return '%s -f -g type -o %s %s' % (os.path.join('.', 'tools', 'generateds', 'generateDS.py'), output, source[0])
```
Comment (katrybacka) [inline rules.py]: same here

**Review by katrybacka — CHANGES_REQUESTED** [review]

**Review by katrybacka — APPROVED** [review]

**Review by sodar — APPROVED** [review]

**Pull request merged**